source/CMakeLists.txt

```diff
@@ -4,13 +4,16 @@ add_library(sqlite3 STATIC ../lib/sqlite3.c)
 find_package(OpenSSL REQUIRED)
 include_directories(${OPENSSL_INCLUDE_DIR})
 
+find_package(Boost REQUIRED)
+include_directories(${Boost_INCLUDE_DIRS})
```

Comment:
I suggest using `target_include_directories` https://cmake.org/cmake/help/latest/command/target_include_directories.html
In modern cmake, we want to have project settings separated.
You need to move it after a definition of a target (after `add_executable`)
Change it in other places as well.